### PR TITLE
Support io direct read cross blocks

### PR DESCRIPTION
**Pull request by LHT129**
issue #40

### CODE REVIEW

File: src/data_cell/flatten_datacell.h [hunk 1]
```diff
@@ -67,7 +67,10 @@ class FlattenDataCell : public FlattenInterface {
     }
 
     [[nodiscard]] const uint8_t*
-    GetCodesById(uint64_t id) const override;
+    GetCodesById(uint64_t id, bool& need_release) const override;
+
+    bool
+    GetCodesById(uint8_t* codes, uint64_t id) const override;
```
Comment (jiaweizone) [inline src/data_cell/flatten_datacell.h]: Adjust the params sequence: Input before output

File: src/io/memory_block_io.h [hunk 2]
```diff
@@ -147,14 +153,16 @@ MemoryBlockIO::ReadImpl(uint8_t* data, uint64_t size, uint64_t offset) const {
 }
 
 const uint8_t*
-MemoryBlockIO::ReadImpl(uint64_t size, uint64_t offset) const {
+MemoryBlockIO::ReadImpl(uint64_t size, uint64_t offset, bool& need_release) const {
     if (check_valid_offset(size + offset)) {
         if (check_in_one_block(offset, size + offset)) {
```
Comment (jiaweizone) [inline src/io/memory_block_io.h:158]: unify name style: checkValidOffset / checkInOneBlock (like memory_io.h:112)

Comment (LHT129) [reply src/io/memory_block_io.h:158]: we use under line style for private function. the memory_io.h will be fixed by other pull request

File: src/data_cell/flatten_datacell.h (hunk 1 above)
Comment (LHT129) [reply src/data_cell/flatten_datacell.h]: done

File: src/data_cell/flatten_datacell.h [hunk 3]
```diff
@@ -165,23 +168,41 @@ FlattenDataCell<QuantTmpl, IOTmpl>::query(float* result_dists,
...
+    }
+
+    return result;
 }
 
 template <typename QuantTmpl, typename IOTmpl>
 const uint8_t*
-FlattenDataCell<QuantTmpl, IOTmpl>::GetCodesById(uint64_t id) const {
-    return io_->Read(code_size_, id * code_size_);
+FlattenDataCell<QuantTmpl, IOTmpl>::GetCodesById(uint64_t id, bool& need_release) const {
```
Comment (jiaweizone) [inline src/data_cell/flatten_datacell.h:199]: add TODO to simplify memory management

**Review by jiaweizone — APPROVED** [review]
LGTM

**Review by wxyucs — APPROVED** [review]
lgtm

**Pull request merged**